setup_root.md

```diff
@@ -1,65 +1,211 @@
 
-### Step 2 (required): Define your data models
+* **Without docker setup:** ./graphql-server/config/data_models_storage_config.json
+* **With docker setup:** ./config/data_models_storage_config.json
+* **SPA in development mode:** ./single-page-app/.env.development
+* **SPA in production mode:** ./single-page-app/.env.production
+* **GraphiQL in development mode:** ./graphql-server/.env.development
+* **GraphiQL in production mode:** ./graphql-server/.env.production
+  If you would like to upload a file to a remote server, please consider the template *.env.migration.sample*, create a new file *.env.migration* and modify relevant environment variables.
 
-To generate model-dependent code that is initially missing within both the graphql-server as well as the single-page-application it is necessary to define your data scheme first. The data scheme can be defined through a set of data models (it is possible to imagine each data model as a separate table in a relation DB). In case you used the CLI (_Step 1_) use the `data_model_definitions` folder in the root directory of your project to define your models.
- <br/>
+If you wish to know more about enviroment variables you can check [this](env_vars.md).
 
- [ > Data Models](setup_data_scheme.md)
-<br/><br/>
+### Step 4: Define your data models
 
-### Step 3 (required): Configure and generate your data models
+Add your model definitions in JSON files to `./data_model_definitions` folder.
 
-After setting up your data models use the Zendro CLI to generate the model-specific code and fill your Zendro skeleton project with life.
+If you want to learn more about how to define data models with Zendro, please check [this](setup_data_scheme.md).
+
+### Step 5: Generate code and migrations
+
+After setting up your data models use the next command to generate the model-specific code and fill your Zendro skeleton project with life.
 
 ```
-zendro generate
+$ zendro generate -m
 ```
 
-This will automatically generate all basic create, read, update and delete (CRUD) operations for each data model specified in the scheme from the previous step. Also have a look at how to generate and use [migrations](https://zendro-dev.github.io/zendro_cli.html#example-for-migrations) using the CLI tool.
+This will automatically generate all basic create, read, update and delete (CRUD) operations for each data model specified in the scheme from the previous step.
 
-You should modify environment variables and database configurations according to your needs.
+Also, this will create migration files. By default, every migration file has two functions, namely up and down. The up function creates a table, the down function deletes the existing table. Furthermore it is possible to customize the migration functions. Please check [this](zendro_cli.md) to learn more about migrations.
 
-### Step 4: Start up your Zendro instance
+***WHEN NOT TO USE -m AND CONNECT TO OWN DB IS MISSING*** 
 
```

Comment:
emphasize on **up** and **down**
***WHEN NOT TO USE -m AND CONNECT TO OWN DB IS MISSING***: actually, the connection can still work if corresponding tables have already been created in the db. The meaning of "-m", is to create migration files. And these files could be executed when graphql-server starts up. That means, we would help to create new tables/indices for data models through migrations. And it is not relevant to the connection. The connection is handled by the `data_models_storage_config.json` file.